### ISSUE TITLE
Geometry intersect results are incorrect if BVH is shallow

### ISSUE THREAD

**Issue by rxmarccall** [gkjohnson/three-mesh-bvh#465]
**Describe the bug**

When trying to do simple Box or Geometry intersect checks, the results show false positives when the objects being compared are roughly on a similar X or Y axis position but are not actually overlapping.

**To Reproduce**

1. Create 2 or more BoxGeometry mesh objects and add them to the Three scene
2. Call `computeBoundsTree` on all Mesh geometries
3. Have one of the meshes follow your mouse for easiest testing
4. Notice that you get false intersect results when positioned at directly above / below, or right / left of other Box Meshes you are comparing against.

***Code***

```js
// This would have reference to the Meshes you've added to the scene and want to check intersection against
const MeshCollection = [];

// Logic for making one Mesh follow the mouse and check intersection against other Meshes in the scene
private onMouseMove = (intersects: THREE.Intersection[]) => {
    if (!intersects.length || !this.mouseFollowMesh) return;

    // Move follow mesh to cursor position in world
    const top = intersects[0];
    this.mouseFollowMesh.position.copy(top.point);
    
    // Check follow Mesh against all others
    const intersectedObjects = checkIntersect(
      this.mouseFollowMesh,
      MeshCollection
    );

    // Set Mesh color depending on intersection result
    MeshCollection.forEach(x => {
      if (intersectedObjects.includes(x)) {
        const mesh = x as THREE.Mesh;
        mesh.material = new THREE.MeshStandardMaterial({ color: 0xff0000 });
      } else {
        const mesh = x as THREE.Mesh;
        mesh.material = new THREE.MeshStandardMaterial({ color: 0x000000 });
      }
    });
}

// The Mesh BVH geometry intersect check logic
export const checkIntersect = (
  shape: THREE.Object3D,
  others: THREE.Object3D[]
): THREE.Object3D[] => {
  const shapeMesh = shape as THREE.Mesh;

  const intersectedObjects: THREE.Object3D[] = [];

  others.forEach(x => {
    // Ignore target intersecting itself
    if (x === shape) return;

    const targetMesh = x as THREE.Mesh;
    if (!targetMesh) return;

    const transformMatrix = new THREE.Matrix4()
      .copy(targetMesh.matrixWorld)
      .invert()
      .multiply(shapeMesh.matrixWorld);

    if (
      targetMesh.geometry.boundsTree?.intersectsGeometry(
        shapeMesh.geometry,
        transformMatrix
      )
    ) {
      intersectedObjects.push(x);
    }
  });

  return intersectedObjects;
};
```

***Live example***

NA

**Expected behavior**

Intersect checks should only return true when the two objects are actually occupying the same 3d space.

**Screenshots**
![bvh_intersects_issue](https://user-images.githubusercontent.com/1920223/191556586-fc0e2e9d-68f5-43e2-b99c-3f775560d9aa.gif)

**Platform:**

 - Device: [Desktop, Mobile, ...]
 - OS: [MacOS]
 - Browser: [Chrome]
 - Three.js version: [r0.132.2]
 - Library version: [v0.4.3]

**Comment by gkjohnson** [comment]
Thanks for the report! Try to reproduce the error using the latest version of the library (v0.5.16) instead of v0.4.3? And if it's still an issue please make a live example with something like a JSFiddle so it's easier to see what's happening.

**Comment by rxmarccall** [comment]
Hey @gkjohnson 
It took me a bit to get an example working in JSFiddle as I haven't worked with it often.
At first when I set it up, it was working correctly and I couldn't get the issue to happen.
After tinkering a bit somehow I got the issue to occur again.

I think I was messing with the cameras sizing values when it started showing the broken behavior, but I don't know for certain.

You can at least see an example where it's acting wacky. Hopefully something will stand out to you.

[JSFiddle](https://jsfiddle.net/ogwju49h/193/)

**Comment by gkjohnson** [comment]
Thanks it looks like this may be a result of an overly simple BVH structure when generating for something like a cube with groups. I'll have to take a deeper dive when I get a chance.

**Comment by rxmarccall** [comment]
I have a few new findings regarding this issue.

It appears that this issue only is occurring with BoxGeometry. I've tried with other Shapes and it's working as expected, even with a Plane Geometry. Around line 50 you can comment / uncomment different shapes and try them out.

Also the 'flickering' of it registering as an intersects is happening because the mouse raycast Z position that gets returned seems to give very small fractional values other than 0.0. I updated the JSFiddle so that the shape's follow Z position is set to 0.0 manually and that is no longer happening.

So now you'll see with BoxGeometry, it's registering intersection all the time. And with other shapes, it's working as expected.

**Comment by rxmarccall** [comment]
And more findings again.

Turns out that BoxGeometry will show correct intersections if you create the geometry and explicitly give it Segment values for width, height, depth of 3 or greater.

I updated the JSFiddle with this finding.

**Comment by gkjohnson** [comment]
I'm suspecting that this now is related to coplanar triangle collision detection. In the jsfiddle it seems to only happen when the boxes are on the same z depth.

**Comment by gkjohnson** [comment]
@rxmarccall this should be fixed in the next release with #489. Thanks for the report!

**Comment by rxmarccall** [comment]
You're welcome, glad it's resolved!
